thavalon-webapp/src/components/Login.tsx

```diff
@@ -3,34 +3,39 @@ import ReactModal from 'react-modal';
 import { useForm } from 'react-hook-form';
 import { Link, Redirect } from 'react-router-dom';
 import { AccountManager } from '../utils/AccountManager';
-import "./Modal.scss";
+import { InputElement } from './formComponents/InputElement';
+import { FormButton } from './formComponents/FormButton';
+import "../styles/Modal.scss";
 
 interface LoginProps {
-    setLoggedIn: React.Dispatch<React.SetStateAction<boolean>>
+    setLoggedIn: React.Dispatch<React.SetStateAction<boolean>>,
+    history: any // it's any because i don't understand error "type 'history' is not a generic"
 };
 
 interface LoginData {
     "email": string,
     "password": string
 };
 
-export function Login(props: LoginProps) {
+export function Login(props: LoginProps): JSX.Element {
     // if set, register modal is open
     const [modalIsOpen, setModalIsOpen] = useState(true);
     // hook for register form
-    const {register, handleSubmit, errors} = useForm<LoginData>();
-    // state for setting if register button is disabled
-    const [disable, setDisabled] = useState(false);
+    const {register, handleSubmit} = useForm<LoginData>();
+    // state for setting if form is being submitted or not
+    const [formSubmitting, setFormSubmitting] = useState(false);
     // state for setting register error
     const [formErrorMsg, setFormErrorMsg] = useState("");
     // state for redirecting to home on successful login
     const [redirectToHome, setRedirectToHome] = useState(false);
 
     /**
-     * Called when register modal is closed.
+     * Called when register modal is closed. Redirects to most recent non-modal page.
      */
     function closeModal() {
         setModalIsOpen(false);
+        // go -1 is equal to going back to prior page, which for login will not be a modal
+        props.history.go(-1);

```

Comment:
General question: does login break if you go back, or are you kept logged in?

Reply:
It shouldn't, but this raises a much better point. If you're logged in, you shouldn't be able to access log in or register pages. Looking at sites like CNN, they do 2 things:

1. Log in does not change URL
2. Going to register page when logged in auto-redirects you to account page

I think I like this behavior better, will look into implementing this. Second should be easily doable, more research needed for first.

Additionally, this has pointed that if you go directly to login or register page when not logged in and attempt to close modal, you get redirected back to the prior site you were on.

Reply:
Added support for #2, although it's a little jank - i had to check the react router, and basically have it attempt to render nothing first because otherwise, it would briefly render register page before redirecting due to async promises